### DISCUSSION TITLE
Generating ddl script

### DISCUSSION THREAD

**Discussion by LucasPenido** [apiman/apiman#1689] [Help] [answered]
Hi!

I am adding a new bean to the `manager/api/beans` module, and I've noticed that the database is being created from the ddls files located in the `distro/data/src/main/resources/ddls` folder and that they are automatically generated by liquibase from files located in `distro/ddl/src/main/liquibase/current`. So I would like to know how to generate the ddl command for my new bean.

**Comment by msavy** [comment, marked as answer]
I've reworked DDL generation slightly on the 3.x branch to make it a bit easier to use, which might give you some inspiration. 

Check out `distro  -> ddl -> pom.xml`

The structure is similar on master/Apiman 2.x , but some sections of the POM are commented out in the older version because there were some dependency issues that came up occasionally and made the DDL generation fail. Feel free to experiment.

For example, on 3.x, you should be able to run `mvn clean install` in the DDL folder and it will generate a new liquibase diff file in `target` that includes your changes. 

Add the generated liquibase diff XML to `src/main/liquibase/current` and modify it to remove anything extraneous  or weird (sometimes liquibase will suggest adding and deleting random unrelated key constraints, so you will want to manually correct that). The other option is just to write your liquibase definition by hand :-).

Then when you run `mvn clean install` again, you will now see updated DDL files in the `target` directory. apiman_h2.ddl etc etc 

Copy these over to `distro -> data -> src -> main -> resources -> ddls`. I strongly suggest using your git diff tool of choice to inspect this and see whether it's generated correctly.

In summary:
- Generate or manually create a liquibase diff as described above. It should be put in `distro/ddl/src/main/liquibase/current` once you are happy with it.
- Regenerate DDL with mvn clean install in the DDL module
- Copy generated DDLs over to `distro/data/src/main/resources/ddls`
- Manually sanity check everything, as sometimes weird things can happen

**Reply by LucasPenido** [reply]
That is what I was looking for
Thanks very much!
